pkg/labels/labels.go

```diff
@@ -143,26 +143,40 @@ type Labels map[string]Label
 
 // GetPrintableModel turns the Labels into a sorted list of strings
 // representing the labels, with CIDRs deduplicated (ie, only provide the most
-// specific CIDR).
+// specific CIDRs).
 func (l Labels) GetPrintableModel() (res []string) {
-	cidr := ""
-	prefixLength := 0
+	// stash away CIDR labels for later de-duplicatin
+	cidrLabels := map[netip.Prefix]Label{}
 	for _, v := range l {
 		if v.Source == LabelSourceCIDR {
-			vStr := strings.Replace(v.String(), "-", ":", -1)
-			prefix := strings.Replace(v.Key, "-", ":", -1)
-			_, ipnet, _ := net.ParseCIDR(prefix)
-			ones, _ := ipnet.Mask.Size()
-			if ones > prefixLength {
-				cidr = vStr
-				prefixLength = ones
+			val := strings.Replace(v.Key, "-", ":", -1)
+			prefix, err := netip.ParsePrefix(val)
+			if err != nil {
+				continue
 			}
-			continue
+			cidrLabels[prefix] = v
+
+		} else {
+			res = append(res, v.String())
 		}
-		res = append(res, v.String())
 	}
-	if cidr != "" {
-		res = append(res, cidr)
+
+	// Remove all prefixes that contain a smaller prefix
+	// Yes, this is O(n^2) but N is ~50
```

Comment:
For one identity. I guess on the commandline we could be printing 1000s of identities? That said as long as it's ideally <100ms even with a large identity list then it's probably in the right order of magnitude for proper interaction with the user. Note this will be hit during sysdump though.

(Would be nice to have a benchmark to evaluate this too..)

Reply:
Good idea. This is actually relevant to more than just `ciliumctl`, as it is also used for generating log messages.

I'll do a quick benchmark.